crates/ruma-common/src/identifiers/room_id.rs

```diff
@@ -26,13 +27,50 @@ pub struct RoomId(str);
 impl RoomId {
     /// Attempts to generate a `RoomId` for the given origin server with a localpart consisting of
     /// 18 random ASCII alphanumeric characters, as recommended in the spec.
+    ///
+    /// This generates a room ID matching the [`RoomIdFormatVersion::V1`] variant of the
+    /// `room_id_format` field of [`RoomVersionRules`]. To construct a room ID matching the
+    /// [`RoomIdFormatVersion::V2`] variant, use [`Self::new_v2()`] instead.
```

Comment:
I know that I wrote this, but we should probably not use `Self` in docs.
```suggestion
    /// [`RoomIdFormatVersion::V2`] variant, use [`RoomId::new_v2()`] instead.
```